source/ui/IDEAS/IDEASXRFDetailedDetectorViewWithSave.cpp

```diff
@@ -253,3 +255,15 @@ void IDEASXRFDetailedDetectorViewWithSave::onDeadTimeCheckButtonClicked()
 
 	deadTimeCheckActions->start();
 }
+
+void IDEASXRFDetailedDetectorViewWithSave::cleanupScanAction()
+{
+	acquireButton_->setEnabled(true);
+	saveScanButton_->setText("Scan Saved");
+	disconnect(scanAction_, SIGNAL(cancelled()), this, SLOT(cleanupScanAction()));
+	disconnect(scanAction_, SIGNAL(succeeded()), this, SLOT(cleanupScanAction()));
+	disconnect(scanAction_, SIGNAL(failed()), this, SLOT(cleanupScanAction()));
+
+	scanAction_->scheduleForDeletion();
+	scanAction_ = 0;
```

Comment:
Because all normal scans (essentially, non-XRF scans) are run through the workflow, their memory is properly handled since the AMActionRunner calls ```scheduleForDeletion()``` on all completed actions.  Then, with my changes with cancel, scans will call their cleanup actions regardless of the final state they arrive in.